pkg/util/framer/framer.go

```diff
@@ -131,13 +131,13 @@ func NewJSONFramedReader(r io.ReadCloser) io.ReadCloser {
 func (r *jsonFrameReader) Read(data []byte) (int, error) {
 	// Return whatever remaining data exists from an in progress frame
 	if n := len(r.remaining); n > 0 {
-		if n <= cap(data) {
+		if n <= len(data) {
```

Comment:
Can you please describe change is this file? I'm not sure I understand them. data is an output parameter here, so why we are checking its initial length?